components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -149,6 +150,7 @@ public void addOAuthApplication(OAuthAppDO consumerAppDO) throws IdentityOAuthAd
                     }
                     addScopeValidators(connection, appId, consumerAppDO.getScopeValidators());
                     // Handle OIDC Related Properties. These are persisted in IDN_OIDC_PROPERTY table.
+                    // TODO : Rename to generic properties for service provider
```

Comment:
Do we need this TODO?
If this is something to be fixed. Better to track it via a git issue

Reply:
I added it to have a discussion and remove it before merging. 

Currently, some methods such as `addServiceProviderOIDCProperties` serve the function of dealing with the configurations that are saved in the database as properties. These properties get saved in **IDN_OIDC_PROPERTY** table. However, all the configurations of a service provider app that is required to be saved as a property might not be OIDC specific. The `AuthorizationCodeValidityPeriod` introduced through this PR is an example. 

Wouldn't it would be better to rename those common methods from OIDC-something to a much generic term such as `addServiceProviderProperties`? 

Please correct me if I am mistaken about this.

Reply:
IMO your concern is valid.

I think the method name more likely indicates where the property ends up being stored (ie. IDN_OIDC_PROPERTY) rather than a classifying them as OIDC related properties. And the properties are not specific to OIDC alone.

I am +1 to rename the method to either addServiceProviderProperties or addOAuthApplicationProperties

Reply:
@mefarazath I am gonna mark this as resolved as we can address the renaming in a different PR if necessary. 
cc. @malithie